### PR TITLE
[MLIR][OpenMP][Docs] NFC: Reorganize 'omp' dialect documentation

### PR DESCRIPTION
**Pull request by skatrak**
This patch creates a handwritten main documentation page for the OpenMP dialect linking to the ODS-generated one as a sub-section.

This new page can be extended to better describe overall design decisions of the dialect rather than relying exclusively on documentation generated automatically from ODS descriptions. After some investigation, there seem to be a few main ways we could structure dialect documentation to allow the introduction of possibly extensive handwritten text.
  - Create a top-level OpenMPDialect.td file that includes the auto-generated one. This is what the `acc` dialect currently does, but it results in the addition of two equal TOCs. It would be possible to move the `include` before all handwritten sections so that the page would have a single TOC, but I believe moving general descriptions to the end of the document would hurt readability. Also keeping the section order without introducing a second TOC would mean the TOC would be inserted somewhere halfway through the page, which isn't useful.
  - Create an OpenMPDialect directory with an _index.md including the auto-generated documentation. This is a different way of reproducing the same issues described above, which is what is currently done for the `linalg` dialect. The multiple TOC issue there is avoided by only including automatically-generated documentation for operations (i.e. `mlir-tblgen -gen-op-doc`) rather than for dialects (i.e. `mlir-tblgen -gen-dialect-doc`). That approach would make it impossible to generate all of the documentation without adding new tablegen backends for `DialectAttr`, `DialectType` and `EnumAttrInfo` definitions or making the TOC optional through a command line option.
  - Create an OpenMPDialect directory with an _index.md that does not include the auto-generated documentation. Instead, link to another document in that directory that includes it. This is the approach taken here, and it circumvents all these issues without having to make any changes to tablegen backends.

### CODE REVIEW

**Comment by skatrak** [comment]
PR Stack:
- #107232
- #107233
- #107234
- #107235
- #107236

File: mlir/docs/Dialects/OpenMPDialect/_index.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# 'omp' Dialect
+
+The `omp` dialect is for representing directives, clauses and other definitions
+of the OpenMP programming model. This directive-based programming model, defined
```
Comment (mjklemm) [inline mlir/docs/Dialects/OpenMPDialect/_index.md:4]: Do you want to leave a pointer to the OpenMP ARB website and maybe even the spec subpage of the OpenMP ARB website?

**Review by mjklemm — APPROVED** [review]
Except for my little nit about referring to the OpenMP website, it LGTM.

File: mlir/docs/Dialects/OpenMPDialect/_index.md (hunk 1 above)
Comment (skatrak) [reply mlir/docs/Dialects/OpenMPDialect/_index.md]: Thanks for the suggestion, done!

**Comment by skatrak** [comment]
Can anyone familiar with the structure of MLIR docs pages take a quick look and let me know if this reorganization is fine? I've been able to build the mlir-www documentation webpage incorporating this change locally and it looks as intended.

**Pull request merged**